polls/tool/src/java/org/sakaiproject/poll/tool/params/PollToolBean.java

```diff
@@ -288,13 +295,93 @@ public String proccessActionAddOption() {
 		manager.saveOption(option);
 		log.debug("Succesuly save option with id" + option.getId());
 
-		if ("option".equals(submissionStatus))
-			return "option";
-		else 
-			return "Saved";
+		String action = "Saved";
+		switch(submissionStatus) {
+			case "option":
+				action = "option";
+				break;
+			case "batch":
+				action = "batch";
+				break;
+			default:
+				break;
+		}
+		return action;
+	}
+
+	public String proccessActionAddOptionBatch() {
+		log.debug("proccessActionAddOptionBatch");
+		if ("cancel".equals(submissionStatus)){
+			log.debug("proccessActionAddOptionBatch: cancel");
+			return "cancel";
+		}
+
+		Long pollId = option.getPollId();
+		if(pollId == null){
```

Comment:
A bit inconsistent on the spaces between if ( and while ( and try (. All that pedantic stuff.

Reply:
Done